df_to_vec.py

```diff
@@ -0,0 +1,199 @@
+import random
+import time
+
+import pandas as pd
+from gensim.models import Word2Vec
+import numpy as np
+import os
+
+WORD_VEC_LENGTH = 100
+NUMBER_OF_TYPES = 1000
+
+w2v_models = {
+    'code': Word2Vec.load('resources/word2vec_model_code.bin'),
+    'language': Word2Vec.load('resources/word2vec_model_language.bin')
+}
+
+
+# NL2Type code
+def vectorize_string(text, feature_length, w2v_model):
+    text_vec = np.zeros((feature_length, WORD_VEC_LENGTH))
+    if text == 'unknown':
+        return text_vec
+    count = 0
+    for word in text.split():
+        if count >= feature_length:
+            return text_vec
+        try:
+            text_vec[count] = w2v_model.wv[word]
+        except KeyError:
+            pass
+        count += 1
+
+    return text_vec
+# /NL2Type code
+
+class Datapoint:
+    def __repr__(self) -> str:
+        values = list(map(lambda kv: kv[0] + ': ' + repr(kv[1]), self.__dict__.items()))
+        values = "\n\t" + ",\n\t".join(values) + "\n"
+        return type(self).__name__ + "(%s)" % values
+
+    def vector_length(self):
+        return sum(self.features.values()) + len(self.features.values()) - 1
+
+    def to_vec(self):
+        datapoint = np.zeros((self.vector_length(), WORD_VEC_LENGTH))
+
+        separator = np.ones(WORD_VEC_LENGTH)
+
+        position = 0
+        for feature, feature_length in self.features.items():
+
+            if self.feature_types[feature] == 'datapoint_type':
+                datapoint[position] = self.datapoint_type_vector()
+                position += 1
+
+            if self.feature_types[feature] == 'code' or self.feature_types[feature] == 'language':
+                vectorized_feature = vectorize_string(
+                    self.__dict__[feature],
+                    feature_length,
+                    w2v_models[self.feature_types[feature]]
+                )
+
+                for word in vectorized_feature:
+                    datapoint[position] = word
+                    position += 1
+
+            # Add separator after each feature
+            if position < len(datapoint):
+                datapoint[position] = separator
+                position += 1
+
+        return datapoint
+
+    def to_be_predicted_to_vec(self):
+        vector = np.zeros(NUMBER_OF_TYPES)
+        # for now a random number
+        vector[random.randint(0, 3)] = 1
+        # vector[self.type] = 1
+        return vector
+
+
+class ParameterDatapoint(Datapoint):
+    def __init__(self, name: str, comment: str, type: int):
+        self.name = name
+        self.comment = comment
+        self.type = type
+
+        self.features = {
```

Comment:
I think the code would be more clear if would at `features_length` (or something similar) instead of `features`.
It reflects a bit better what these values represents